javatests/arcs/android/e2e/testapp/TestActivity.kt

```diff
@@ -431,33 +362,24 @@ class TestActivity : AppCompatActivity() {
             )
             startService(intent)
         } else {
-            singletonHandle?.dispatcher?.let {
-                withContext(it) { singletonHandle?.clear()?.join() }
-            }
-            collectionHandle?.dispatcher?.let {
-                withContext(it) { collectionHandle?.clear()?.join() }
-            }
+            singletonHandle?.dispatchClear()
+            collectionHandle?.dispatchClear()
         }
     }
 
     private fun fetchAndUpdateResult(prefix: String) {
-        var result: String? = "null"
+        var result = "null"
         if (isCollection) {
-            collectionHandle?.dispatcher?.let {
-                val testEntities = runBlocking(it) { collectionHandle?.fetchAll() }
-                if (testEntities != null && !testEntities.isNullOrEmpty()) {
-                    result = testEntities
-                        .sortedBy { it.number }
-                        .joinToString(separator = ";") { "${it.text},${it.number},${it.boolean}" }
-                }
+            //~ runBlocking(dispatcher) ?
```

Comment:
And here?

Reply:
Similar; wasn't sure about needing to runBlock on the dispatcher. I'm fairly sure it should be fine not too.